mobile/src/components/courses/components/category-list/category-list.tsx

```diff
@@ -9,48 +9,38 @@ import { styles } from './style';
 
 type Props = {
   items: CategoryGetAllItemResponseDto[];
+  handleSelect: (id: number) => void;
+  activeCategoryId: number | undefined;
 };
 
-const CategoryList: FC<Props> = ({ items }) => {
-  const [activeCategoryIds, setActiveCategoryIds] = useState<number[]>([]);
+const CategoryList: FC<Props> = ({ items, handleSelect, activeCategoryId }) => {
   const [filteredCategories, setFilteredCategories] = useState<
     CategoryGetAllItemResponseDto[]
   >([]);
 
-  const handlePress = (id: number): void => {
-    if (!activeCategoryIds.includes(id)) {
-      setActiveCategoryIds([...activeCategoryIds, id]);
-    } else {
-      setActiveCategoryIds(
-        activeCategoryIds.filter((activeId) => activeId !== id),
-      );
-    }
-  };
-
   useEffect(() => {
-    if (activeCategoryIds.length) {
-      const activeSet = new Set(activeCategoryIds);
-      const activeItems = items.filter((item) => activeSet.has(item.id));
-      const notActiveItems = items.filter((item) => !activeSet.has(item.id));
-
-      setFilteredCategories([...activeItems, ...notActiveItems]);
+    if (activeCategoryId) {
+      const activeItem = items.filter((item) => item.id === activeCategoryId);
+      const notActiveItems = items.filter(
+        (item) => item.id !== activeCategoryId,
+      );
+      setFilteredCategories([...activeItem, ...notActiveItems]);
```

Comment:
To avoid such modifications, let's replace ScrollView with FlatList and use it's scrollToIndex method, it will be more user friendly

Reply:
Please describe, how it should work

Reply:
Because, as I think, if we won't move selected category to the far left edge, we don't need scrollToIndex at all

Reply:
Actually, we don`t need to move selected category to left edge, just scroll into view area